fix: Improve node ID check robustness & WebdriverIO test CI stability

BenHenning opened the pull request:
https://github.com/google/blockly/pull/9384 revealed that one of the Block comment tests was a bit fragile in directly validating a specific node ID. Since node IDs can change if new elements on the page require a generated ID, the test could easily fail with changes to core Blockly.

This PR fixes the issue by using a regex check instead of a direct ID match. The 'comment_textarea' part is _probably_ sufficiently unique for the test to be considered correct (though it is possible for the wrong comment to be focused here, but it seems really unlikely with how simple the test is being set up and, if that actually happened, it's sort of incidentally still correct as there would be _some_ comment that has focus).

This PR also includes a number of attempts to improve test stability (particularly in CI):
- It introduces a CI-specific command that adds a 30 second Mocha test timeout to provide a lot more time for tests to fail (since CI tends to run on less performant machines than used for development).
- It introduces a synchronization between the Mocha timeout and the timeout used for WebdriverIO's `waitFor*` functions. Note that this actually will usually be 60 seconds regardless of the configured Mocha timeout due to the way the hook is set up. However, this is still far better than the 5 second default as at least 1 test seemed to fairly regularly fail against that limit.
- It adds a few missing 'wait' calls for helpers that are performing side effects which ensures better synchronization when using `PAUSE_TIME` (and possibly better stability when running in CI, too, though it's unclear of `pause()` actually does anything when the provided timeout is 0).
- See https://github.com/google/blockly-keyboard-experimentation/pull/745#issuecomment-3336166191 attempts to further stabilize OS X tests only to yield that there are actual test failures.

This will be merged into the screen reader experimental branch to unblock the core Blockly PR as well.

#746 has been filed to track the discovered test failures.

BenHenning committed:
fix: Improve node ID check robustness

gonfunko reviewed (approved):
LGTM once the lint failure is resolved.

BenHenning commented on test/webdriverio/test/block_comment_test.ts:
```diff
@@ -36,7 +36,8 @@ suite('Block comment navigation', function () {
     const focusedNodeId = await getCurrentFocusNodeId(this.browser);
-    chai.assert.equal(focusedNodeId, 'blockly-2s_comment_textarea_');
+    chai.assert.isDefined(focusedNodeId);
+    chai.assert.match(focusedNodeId, /blockly\-\w+_comment_textarea_/);
```
```suggestion
    chai.assert.match(focusedNodeId, /blockly-\w+_comment_textarea_/);
```

Addresses lint failure.

BenHenning committed:
fix: Lint issue.

BenHenning committed:
feat: Support CI-specific WDIO runs.

This adds a 30 second timeout for WDIO tests, though it's not yet enabled on CI.

BenHenning committed:
feat: Use extended Mocha timeouts in CI.

BenHenning committed:
Fix: Fix package.json formatting issue.

BenHenning committed:
feat: Attempt to improve timeouts.

This adds more waits and ensures that WebdriverIO timeouts synchronize
with Mocha's (for better CI stability).

BenHenning committed:
feat: Attempt to slow down OS X tests.

BenHenning committed:
Revert "feat: Attempt to slow down OS X tests."

This reverts commit ae92cf5abd2fc0a8f0db716c3ad01c8dd573a092.
